### ISSUE TITLE
an asynchronous version of the v2 API wrapper

### ISSUE THREAD

**Issue by bemxio** [circleguard/ossapi#44]
it would be quite useful for website backends, with logins and discord bots, that also want to authenticate users in a way
https://async-oauthlib.readthedocs.io/en/latest/ there's an async oauth module, that seems to fit the features needed for it
and for requests themself, `aiohttp` would be nice
it could be named something like `AsyncOssapiV2` which would be used the same as the synchronous one, except that all endpoints would need to be `await`ed

**Comment by tybug** [comment]
yep, this has been on my mind for a while now. I'll write an async version at some point.

**Comment by xFGhoul** [comment]
possibly separate them into folders so imports become like `sync.OssapiV2` or `async.OssapiV2`

**Comment by tybug** [comment]
I'd prefer if the async version was called `AsyncOssapiV2` to clearly mark it as async, and the sync version continue to be the default `OssapiV2`. 

I did start working on this, and you can see my (extremely rough and currently erroring) progress on the async branch, namely commit https://github.com/circleguard/ossapi/commit/b420040989fc8b9982989355b0fba71f607335f6.

It turned out to be a huge pain to extract out the async portions of the code cleanly. Finishing this would likely require another 10-20 hours of work, and I'm also currently blocked by https://github.com/weibeu/Async-OAuthlib/issues/3 for async oauth. The author of that library appears to be inactive, so I'd likely need to either fix it myself or roll my own implementation. I would much prefer not doing the latter, since we already have clean integration with oauthlib's `Session` class, which async-oauthlib mirrors.

In short, don't expect this anytime soon, since I won't have much time to work on it again until the summer.

**Comment by tybug** [comment]
I've implemented this in v3.1.0. You can import it as `OssapiAsync` and use it identically to `Ossapi`. Read more on the docs: https://circleguard.github.io/ossapi/async.html.
